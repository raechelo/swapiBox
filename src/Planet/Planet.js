import React from 'react';

const Planet = (props) => {
  return (
    <article className="Card">
      <h4>{props.name}</h4>
      <h6>Population: {props.population}</h6>
      <h6>Terrain: {props.terrain}</h6>
      <h6>Climate: {props.climate}</h6>
      <h6>Residents: {props.residents}</h6>
      <h6><i class="far fa-star"></i></h6>      
    </article>
  )
}

export default Planet;