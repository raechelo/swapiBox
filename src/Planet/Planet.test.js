import React from 'react';
import { shallow } from 'enzyme';
import Planet from './Planet';

describe('Planet', () => {

  let wrapper;

  let mockName = 'Mordor';
  let mockPopulation = 299999;
  let mockTerrain = 'rocky';
  let mockResidents = ['Sauron'];
  let mockClimate = 'burning hot';

  beforeEach(() => {
    wrapper = shallow (
      <Planet name={mockName} population={mockPopulation} terrain={mockTerrain} residents={mockResidents} climate={mockClimate} />
    )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot in absence of data', () => {
    mockResidents = [];
    expect(wrapper).toMatchSnapshot();
  })
})