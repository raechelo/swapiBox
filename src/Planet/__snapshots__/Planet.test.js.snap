// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Planet should match the snapshot in absence of data 1`] = `
<article
  className="Card"
>
  <h4>
    Mordor
  </h4>
  <h6>
    Population: 
    299999
  </h6>
  <h6>
    Terrain: 
    rocky
  </h6>
  <h6>
    Climate: 
    burning hot
  </h6>
  <h6>
    Residents: 
     -Sauron
  </h6>
  <h6>
    <i
      class="far fa-star"
      onClick={[Function]}
    />
  </h6>
</article>
`;

exports[`Planet should match the snapshot with all the data passed in 1`] = `
<article
  className="Card"
>
  <h4>
    Mordor
  </h4>
  <h6>
    Population: 
    299999
  </h6>
  <h6>
    Terrain: 
    rocky
  </h6>
  <h6>
    Climate: 
    burning hot
  </h6>
  <h6>
    Residents: 
     -Sauron
  </h6>
  <h6>
    <i
      class="far fa-star"
      onClick={[Function]}
    />
  </h6>
</article>
`;
